app/models/comment.rb

```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :user, foreign_key: 'author_id'
```

Comment:
it's working as expected but the way that you used it may be redundant if large code base so it would be better in the future to use `belongs_to :author, class_name: 'User'`